cmake/multithreading.cmake

```diff
@@ -51,7 +51,7 @@ endif()
 
 if (TBB_FOUND)
   INCLUDE_DIRECTORIES(${TBB_INCLUDE_DIRS})
-  add_compile_options(/DOPENMS_HAS_TBB)
+  add_compile_options(-DOPENMS_HAS_TBB)
```

Comment:
fixes #3382

Reply:
I did look into whether the TBB data structures did give any benefit but I did not pursue this for this one reasons:

- we still need the singleton access, which means that we dont really want concurrent insertions and reads, rather we want to have sequential insertions and reads. Otherwise we would have multiple entries of the same residue in the singleton etc...
- TBB is great if you need to insert different element from multiple threads but not if you are potentially inserting the same element, there is no other way around this
- TBB could still help by making the check whether the element exists "atomic" and only then go into the read loop. I have not tested this so far
- all the code does compile right now for Residue and ModificationsDB if you simply swap out the OpenMS data structures with TBB concurrent data structures (this is why all the `auto` is now in the code to make it work seamlessly)